### PR TITLE
Myavatar targetvelocity

### PR DESCRIPTION
**Pull request by Triplelexx**
Attempt at exposing updated controller's avatar velocity calculations to scripts.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
An admin must verify this patch for the pull request build to fire. Type "ok to test" to begin testing this pull request or "add to whitelist" to test this and all future pull requests from this collaborator.

File: interface/src/avatar/MyAvatar.cpp [hunk 1]
```diff
@@ -1240,28 +1240,28 @@ glm::vec3 MyAvatar::applyScriptedMotor(float deltaTime, const glm::vec3& localVe
...
     bool isHovering = _characterController.isHovering();
     glm::vec3 newLocalVelocity = applyKeyboardMotor(deltaTime, localVelocity, isHovering);
     newLocalVelocity = applyScriptedMotor(deltaTime, newLocalVelocity);
 
     // rotate back into world-frame
-    _velocity = rotation * newLocalVelocity;
+    _targetVelocity = rotation * newLocalVelocity;
 
-    _velocity += _thrust * deltaTime;
+    _targetVelocity += _thrust * deltaTime;
```
Comment (PhilipRosedale) [inline interface/src/avatar/MyAvatar.cpp]: I don't understand - interpolation should be based on actual, not target velocity.  I see the concept of a target velocity being introduced, but I the measured/actual velocity needs to be used through all this code.

**Comment by PhilipRosedale** [comment]
Don't understand how this would work - can you write up a description of how we would test/use this change?

**Comment by AndrewMeadows** [comment]
Right now in upstream/master the AvatarData::_velocity field is being used to store the "target velocity" of the avatar, rather than its true velocity.  Which means when scripts use MyAvatar.velocity they aren't getting the right value.  In order to store the true velocity in this data field while keeping the current avatar motion unchanged we need to create a MyAvatar::_targetVelocity field so we can free up AvatarData::_velocity field.

The idea of the "target velocity" is that the avatar has a motor that spins up to some maximum speed, but it takes some time to get there.  Once it has spun up it stays spinning, even if it is not actually getting traction (pushing against a wall, for example).  

In short: we may want to change how the "target velocity" system works, but in order to get JS access to the true avatar velocity without breaking stuff we'll need Triplelexx's changes.

**Comment by Triplelexx** [comment]
Yes this change was just made in an attempt at getting the true velocity in JS. In testing I have found that the change doesn't appear to drastically effect smooth movement, or collision, even in Debug. Yet will provide accurate tracking information to JS.

**Comment by PhilipRosedale** [comment]
OK to test

**Comment by AndrewMeadows** [comment]
test this please

**Comment by AndrewMeadows** [comment]
I cloned the repo and tested.  It works the same as far as I can tell.
